pingcap/tidb
# executor: Add plugin column to mysql.user

### What problem does this PR solve?

Problem Summary:

This is to prepare for other authentication plugins besides `mysql_native_password`.

### What is changed and how it works?

This adds a `mysql.user.plugin` column. For now this would always contain `mysql_native_password`, but later on it could contain `caching_sha2_password` for #9411 

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Manual tests:
- Upgrade (bootstrap code to add the column and set a value)
- Create a new user (which should fill in the column)
- use `SHOW CREATE USER` which should include the auth plugin
- Check mysql.user.plugin for each of these.

### Release note 

- No release note
